.github/workflows/main.yml

```diff
@@ -56,7 +56,14 @@ jobs:
         run: |
           cd demo
           python manage.py makemigrations
-          coverage run manage.py test plans
+          case "${{ matrix.DJANGO_VERSION }}" in
+            2.2.*|3.0.*|3.1.*)  # Django runs with warnings in this python version
+              python -m coverage run manage.py test plans
+              ;;
+            *)
+              python -W error::DeprecationWarning -m coverage run manage.py test plans
+              ;;
+          esac
```

Comment:
@pandafy Why do you think so?

If there is deprecation warning on Django/Python it would not appear on the versions pinned in the workflow. So only if someone decides to add new version, than it will fail. Which is very desirable in my opinion, because it will tell him to fix deprecation in advance so when the next version is out, everything will be solved.

There is slight risk, that deprecation warning can appear in some dependent library, which is not pinned. Then the correct MRs would fail. But I think, the risk is very little, because there are only few dependencies and they are developing only very slowly.

Reply:
Okay 👍🏼